front/lib/plans/subscription.ts

```diff
@@ -0,0 +1,171 @@
+import { front_sequelize, ModelId } from "@app/lib/databases";
+import { Plan, Subscription } from "@app/lib/models";
+import {
+  FREE_TRIAL_PLAN_CODE,
+  TEST_PLAN_CODE,
+} from "@app/lib/plans/free_plans";
+import { generateModelSId } from "@app/lib/utils";
+import logger from "@app/logger/logger";
+import { SubscribedPlanType } from "@app/types/user";
+
+export const getActiveWorkspacePlan = async ({
+  workspaceModelId,
+}: {
+  workspaceModelId: ModelId;
+}): Promise<SubscribedPlanType> => {
+  let plan = null;
+  let startDate = new Date();
+  let endDate = null;
+
+  const activeSubscription = await Subscription.findOne({
+    where: { workspaceId: workspaceModelId, status: "active" },
```

Comment:
I find it weird to have both a static "status" string, and the start/end date. How do we make sure this doesn't go stale / out of sync ? Can't the status be derived from the dates instead ?

Reply:
We have a db hook constraint making sure there's only one `isActive` for a given workspace, and the dates are here just for posterity. It makes it easy to fetch the current Subscription (find one where status = isActive) over having to search among the dates (findAll where today is between start and end date) while making sure there is only one result. 

WDYT?

Reply:
I am personally against duplicating information in the database. Anything that _can_ go out of sync inevitably _will_ go out of sync.
There can only be a single `isActive`, but if there's an end date to the latest subscription has an end date, there's nothing that will come and set it to "ended". I guess this is somewhat related to the conversation of having "test plan" subscriptions in the DB or not.

> It makes it easy to fetch the current Subscription

Reply:
Got to leave early today but will address tomo morning 🙏🏻

Reply:
Let's consider the status as the ground turth and the end date as an audit log. Managing which is the active subscription by date is going to be horrible, if only to quickly debug things in metabase.